### PR TITLE
Don't ask for 'rate signal on play store' on the wesite release

### PR DESCRIPTION
**Pull request by Leptopoda**
### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * AVD, ANdroid 11
 * Xiaomi mi A3, Android 10 (LOS 17.0)
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/) [brought up on the forum](https://community.signalusers.org/t/disable-rate-signal-banner-on-non-play-service-devices/17313)

----------

### Description

On devices using the website release users should not be asked to rate the app on the play store.
Also the `RatingManger` does now use the `PlayStoreUtil` removing duplicate code and better handling if the play store is not present (by opening the browser).
The screen is currently still displayed on non play services devices thus not affecting the `playProdRelease`

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/components/RatingManager.java [hunk 1]
```diff
@@ -25,6 +27,11 @@
 
   public static void showRatingDialogIfNecessary(Context context) {
     if (!TextSecurePreferences.isRatingEnabled(context)) return;
```
Comment (Leptopoda) [inline app/src/main/java/org/thoughtcrime/securesms/components/RatingManager.java:29]:
```suggestion
    if (!TextSecurePreferences.isRatingEnabled(context) || BuildConfig.PLAY_STORE_DISABLED) return;
```
ok, don't know what the hack i was thinking but the new check should be integrated with the other one

Comment (Leptopoda) [reply app/src/main/java/org/thoughtcrime/securesms/components/RatingManager.java]: just made the change (though haven't tested it yet)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1327,7 +1327,6 @@
     <string name="RatingManager_rate_now">Rate now!</string>
     <string name="RatingManager_no_thanks">No thanks</string>
     <string name="RatingManager_later">Later</string>
-    <string name="RatingManager_whoops_the_play_store_app_does_not_appear_to_be_installed">Whoops, the Play Store app does not appear to be installed on your device.</string>
```
Comment (bitmold) [inline app/src/main/res/values/strings.xml:1330]: You can run Android lint to find every instance of this (now unused) resource in other locales and remove them all

Comment (Leptopoda) [reply app/src/main/res/values/strings.xml:1330]: Yeah indeed I could but judging from the past after two syncs from transifex the strings will be removed automatically.

One sync to upload the deletion to transifex and another one to get the deletion of the unused stings into android again

**Comment by greyson-signal** [comment]
I've put this in the upcoming 5.5.2 branch, thanks!

**Pull request closed without merging**